assignment-client/src/assets/AssetServer.cpp

```diff
@@ -371,8 +412,86 @@ void AssetServer::sendStatsPacket() {
     ThreadedAssignment::addPacketStatsAndSendStatsPacket(serverStats);
 }
 
-void AssetServer::loadMappingFromFile() {
+static const QString MAP_FILE_NAME = "map.json";
+
+void AssetServer::loadMappingsFromFile() {
+
+    auto mapFilePath = _resourcesDirectory.absoluteFilePath(MAP_FILE_NAME);
+
+    QFile mapFile { mapFilePath };
+    if (mapFile.exists()) {
+        if (mapFile.open(QIODevice::ReadOnly)) {
+            auto jsonDocument = QJsonDocument::fromJson(mapFile.readAll());
```

Comment:
If this fails we will (presumably) load an empty map, and possibly overwrite it in the future. We should at least check for errors and possibly bail if there is one.